client/trino-client/src/test/java/io/trino/client/TestResultRowsDecoder.java

```diff
@@ -292,8 +285,14 @@ private static QueryResults fromSegments(Segment... segments)
                 .build());
     }
 
-    private static Segment spooledSegment()
+    private static Segment spooledSegment(long rows)
     {
-        return spooled(URI.create("http://localhost"), URI.create("http://localhost"), DataAttributes.empty(), ImmutableMap.of());
+        DataAttributes attributes = DataAttributes.builder()
+                .set(DataAttribute.ROWS_COUNT, rows)
+                .set(DataAttribute.ROW_OFFSET, -1L)
+                .set(DataAttribute.SEGMENT_SIZE, -1)
+                .build();
```

Comment:
unrelated

Reply:
Related. At least rows count (needed by iterator for sanity checks)

Reply:
where did the logic change so it is now needed and was not before?

Reply:
in the iterator which checks for the progression of the rows vs the total count of rows